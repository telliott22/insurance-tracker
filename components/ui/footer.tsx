import React from "react";
import { Shield } from "lucide-react";

export function Footer() {
  return (
    <footer className="border-t border-slate-700 bg-slate-900">
      <div className="max-w-7xl mx-auto py-12 px-4 sm:px-6 lg:px-8">
        <div className="flex flex-col md:flex-row justify-between items-center">
          <div className="flex items-center space-x-2 mb-4 md:mb-0">
            <Shield className="h-6 w-6 text-blue-500" />
            <span className="text-lg font-semibold text-white">Insurance Tracker</span>
          </div>
          <div className="flex flex-col items-center gap-2 text-slate-400 text-sm">
            <p>Built with Next.js, Supabase, and OpenAI</p>
            <p>Made with ❤️ in Berlin</p>
          </div>
        </div>
      </div>
    </footer>
  );
}
