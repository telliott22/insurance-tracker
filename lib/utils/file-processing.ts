/**
 * Utility functions for file processing and conversion
 */

/**
 * Convert a File to base64 string
 */
export async function fileToBase64(file: File): Promise<string> {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => {
      const result = reader.result as string;
      // Remove the data URL prefix (e.g., "data:image/jpeg;base64,")
      const base64 = result.split(',')[1];
      resolve(base64);
    };
    reader.onerror = (error) => reject(error);
  });
}

/**
 * Convert PDF to images using canvas (client-side)
 * Note: This is a placeholder - for production you might want to use pdf-lib or similar
 */
export async function pdfToImages(file: File): Promise<string[]> {
  // For now, return the PDF as base64 - OpenAI can handle PDF files directly
  const base64 = await fileToBase64(file);
  return [base64];
}

/**
 * Validate file type and size
 */
export function validateFile(file: File): { valid: boolean; error?: string } {
  const allowedTypes = [
    'image/jpeg',
    'image/jpg', 
    'image/png',
    'application/pdf'
  ];

  const maxSize = 10 * 1024 * 1024; // 10MB

  if (!allowedTypes.includes(file.type)) {
    return {
      valid: false,
      error: 'Invalid file type. Only JPEG, PNG, and PDF files are supported.'
    };
  }

  if (file.size > maxSize) {
    return {
      valid: false,
      error: 'File too large. Maximum size is 10MB.'
    };
  }

  return { valid: true };
}

/**
 * Generate a unique filename
 */
export function generateUniqueFilename(originalName: string, userId: string): string {
  const timestamp = Date.now();
  const randomSuffix = Math.random().toString(36).substring(2, 8);
  const extension = originalName.split('.').pop();
  return `${userId}/${timestamp}_${randomSuffix}.${extension}`;
}

/**
 * Format file size for display
 */
export function formatFileSize(bytes: number): string {
  if (bytes === 0) return '0 Bytes';

  const k = 1024;
  const sizes = ['Bytes', 'KB', 'MB', 'GB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  return parseFloat((bytes / Math.pow(k, i)).toFixed(2)) + ' ' + sizes[i];
}

/**
 * Check if file is an image
 */
export function isImageFile(file: File): boolean {
  return file.type.startsWith('image/');
}

/**
 * Check if file is a PDF
 */
export function isPDFFile(file: File): boolean {
  return file.type === 'application/pdf';
}

/**
 * Create a preview URL for images
 */
export function createPreviewUrl(file: File): string | null {
  if (isImageFile(file)) {
    return URL.createObjectURL(file);
  }
  return null;
}

/**
 * Cleanup preview URLs to prevent memory leaks
 */
export function cleanupPreviewUrl(url: string): void {
  if (url && url.startsWith('blob:')) {
    URL.revokeObjectURL(url);
  }
}