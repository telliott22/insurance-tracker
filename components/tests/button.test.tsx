import React from 'react'
import { render, screen } from '../../__tests__/utils/test-utils'
import { Button } from '../ui/button'

describe('Button', () => {
  it('should render button with text', () => {
    render(<Button>Click me</Button>)
    expect(screen.getByRole('button', { name: 'Click me' })).toBeDefined()
  })

  it('should apply variant classes', () => {
    render(<Button variant="destructive">Delete</Button>)
    const button = screen.getByRole('button')
    expect(button.className).toContain('bg-destructive')
  })

  it('should apply size classes', () => {
    render(<Button size="lg">Large Button</Button>)
    const button = screen.getByRole('button')
    expect(button.className).toContain('h-10')
    expect(button.className).toContain('px-8')
  })

  it('should be disabled when disabled prop is true', () => {
    render(<Button disabled>Disabled</Button>)
    const button = screen.getByRole('button') as HTMLButtonElement
    expect(button.disabled).toBe(true)
  })

  it('should match snapshot for default variant', () => {
    const { container } = render(<Button>Default Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should match snapshot for secondary variant', () => {
    const { container } = render(<Button variant="secondary">Secondary Button</Button>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
