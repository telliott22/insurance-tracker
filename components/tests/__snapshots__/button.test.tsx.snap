// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot for default variant 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
>
  Default Button
</button>
`;

exports[`Button should match snapshot for secondary variant 1`] = `
<button
  class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80 h-9 px-4 py-2"
>
  Secondary Button
</button>
`;
